mesonbuild/build.py

```diff
@@ -1120,10 +1053,14 @@ def process_kwargs(self, kwargs):
         self.add_deps(deplist)
         # If an item in this list is False, the output corresponding to
         # the list index of that item will not be installed
+        #
+        # Yes, this is allowed to be a list of str | bool, because of
+        # vala targets.
+        # TODO: use explicit keyword arguments instead
         self.install_dir = typeslistify(kwargs.get('install_dir', []),
                                         (str, bool))
         self.install_mode = kwargs.get('install_mode', None)
-        self.install_tag = stringlistify(kwargs.get('install_tag', [None]))
+        self.install_tag = _process_install_tag([kwargs.get('install_tag')], 1)
```

Comment:
Wondering if this has always been wrong. As you pointed in the comment above, there could be more than one output because of vala. Those outputs needs a tag too. In backend.generate_target_install() it asserts that len(self.install_tag) == len(t.get_outputs()). Wondering why we never hit that assert, we don't have vala unit tests? Maybe something I don't understand.

Reply:
We hardcode tags for the extra vala outputs

Reply:
I see, we have no other choice because that's where we add the extra outputs. All good then, thanks.